app/application/use_cases/documents/document_operations.py

```
@@ -41,12 +41,26 @@ def _rewind_if_seekable(file_data: BinaryIO) -> None:
+
 def _sanitize_filename(filename: str) -> str:
-    """Strip path separators and dangerous characters from filename."""
+    """Strip path separators and dangerous characters from filename.
+
+    Rejects reserved names (e.g. CON, NUL on Windows) to avoid storage issues
+    when using local or Windows-accessible storage.
+    """
     name = os.path.basename(filename)
     name = name.replace("\x00", "").strip(". ")
     if not name:
         raise ValueError("Filename is empty or invalid after sanitization")
```

Ruff (0.15.2)

[warning] Avoid specifying long messages outside the exception class

(TRY003)